providers/asrockrack/firmware.go

```diff
@@ -94,112 +101,94 @@ func (a *ASRockRack) firmwareInstallBIOS(ctx context.Context, reader io.Reader,
 	if err != nil {
-		return fmt.Errorf("failed in step 3/3 - proceed with upgrade: " + err.Error())
+		return errors.Wrap(err, "failed in step 3/3 - proceed with upgrade")
 	}
 
 	return nil
 }
 
-// firmwareUpdateBMCStatus returns the BMC firmware install status
-func (a *ASRockRack) firmwareUpdateBMCStatus(ctx context.Context, installVersion string) (status string, err error) {
-	p, progressErr := a.flashProgress(ctx, "api/maintenance/firmware/flash-progress")
-	if progressErr != nil {
-		installed, versionErr := a.versionInstalled(ctx, devices.SlugBMC, installVersion)
-		if err != nil {
-			return "", versionErr
-		}
-
-		if installed {
-			return devices.FirmwareInstallComplete, nil
-		}
-
-		return "", progressErr
-	}
+// firmwareUpdateBIOSStatus returns the BIOS firmware install status
+func (a *ASRockRack) firmwareUpdateStatus(ctx context.Context, component string, installVersion string) (status string, err error) {
+	// TODO: purge debug logging
+	os.Setenv("BMCLIB_LOG_LEVEL", "trace")
+	defer os.Unsetenv("BMCLIB_LOG_LEVEL")
 
-	switch p.State {
-	case 0:
-		return devices.FirmwareInstallRunning, nil
-	case 2:
-		return devices.FirmwareInstallComplete, nil
+	var endpoint string
+	switch component {
+	case devices.SlugBIOS:
+		endpoint = "api/asrr/maintenance/BIOS/flash-progress"
+	case devices.SlugBMC:
+		endpoint = "api/maintenance/firmware/flash-progress"
 	default:
-		a.log.V(3).Info("warn", "bmc returned unknown flash progress state: "+strconv.Itoa(p.State))
+		return "", errors.Wrap(bmclibErrs.ErrFirmwareInstallStatus, "component unsupported: "+component)
 	}
 
-	// at this point the flash-progress endpoint isn't returning useful information
-	// query the fimware info endpoint to identify the installed version
-	installed, err := a.versionInstalled(ctx, devices.SlugBMC, installVersion)
+	// 1. query the flash progress endpoint
+	//
+	// once an update completes/fails this endpoint will return 500
+	progress, err := a.flashProgress(ctx, endpoint)
 	if err != nil {
-		return "", err
+		a.log.V(3).Info("warn", "bmc query for install progress returned error: "+err.Error())
 	}
 
-	if installed {
-		return devices.FirmwareInstallComplete, nil
-	}
-
-	return devices.FirmwareInstallUnknown, nil
-}
-
-// firmwareUpdateBIOSStatus returns the BIOS firmware install status
-func (a *ASRockRack) firmwareUpdateBIOSStatus(ctx context.Context, installVersion string) (status string, err error) {
-	p, progressErr := a.flashProgress(ctx, "api/asrr/maintenance/BIOS/flash-progress")
-	if progressErr != nil {
-		installed, versionErr := a.versionInstalled(ctx, devices.SlugBIOS, installVersion)
-		if versionErr != nil {
-			return "", versionErr
-		}
-
-		if installed {
+	if progress != nil {
+		switch progress.State {
+		case 0:
+			return devices.FirmwareInstallRunning, nil
+		case 2:
 			return devices.FirmwareInstallComplete, nil
+		default:
+			a.log.V(3).Info("warn", "bmc returned unknown flash progress state: "+strconv.Itoa(progress.State))
```

Comment:
Since you're emitting structured logs, maybe put the state in a json field?
```suggestion
			a.log.V(3).WithValues("state", pgorgess.State).Info("bmc returned unknown flash progress state")
```